API.md

```diff
@@ -0,0 +1,82 @@
+# CRUD ROUTES
+
+## Create (POST)
+
+Post one song to songs table in database;
+
+`POST /songs/`
+
+Parameters
+
+| Name | Type | Description |
+|:----:|:----:|:----:|
+| `title` | `text` | *Required.* Song title. |
+| `album` | `text` | *Required.* Album song comes from. |
+| `artist` | `text` | *Required.* Name of artist. |
+| `likes` | `integer` | *Required.* Number of likes song has recieved. |
+| `length` | `integer` | *Required.* Total length of song. |
+| `upNext` | `text` | *Required.* Song to be played next. |
+| `prevPlayed` | `text` | *Required.* Previously played song. |
+
+
+## Read (GET)
+
+Get a single song by the song id.
+
+`GET /songs/:id`
+
+Parameters
+
+| Name | Type | Description |
+|:----:|:----:|:----:|
+| `songId` | `integer` | *Required.* Id of song in database. |
+
+Response
+
+| Name | Type | Description |
+|:----:|:----:|:----:|
+| `songId` | `integer` | *Required.* Song id of selected song. |
+| `title` | `text` | *Required.* Title of song. |
+| `album` | `text` | *Required.* Name of album song belongs to. |
+| `artist` | `text` | *Required.* Name of artist. |
+| `upNext` | `text` | Name of song to be played next. |
+| `prevPlayed` | `text` | Name of previously played song. |
+| `length` | `decimal` | *Required.* Length of song in minutes and seconds. |
+| `likes` | `integer` | *Required.* Number of likes for this song. |
+| `timestamp` | `decimal` | Timestamp of last playing song. |
+
+## Update (PATCH)
+
+Add or subtract a like from a specific song
+
+`PATCH /likes/:songId`
+
+Parameters
+
+| Name | Type | Description |
+|:----:|:----:|:----:|
+| `songId` | `integer` | *Required.* Id of current song. |
+| `likes` | `integer` | *Required.* The number 1 for a single like. |
+
+Save the timestamp of the current playing song
+
+`PATCH /songs/:songId/:timestamp`
+
+Parameters
+
+| Name | Type | Description |
+|:----:|:----:|:----:|
+| `songId` | `integer` | *Required.* Id of current song. |
+| `timestamp` | `decimal` | *Required.* Timestamp of currently playing song in minutes and seconds. |
```

Comment:
Not sure if decimal is the correct type of data for a timestamp. Mixing decimals and time will get confusing.